webapp/src/client/client.js

```diff
@@ -263,68 +264,76 @@ export default class Client {
         };
     };
 
-    doGet = async (url, headers = {}, credentials) => {
-        headers.Accept = 'application/json';
-        let options = {
-            method: 'get',
-            headers,
-        };
-
-        if (url.includes('plugins/' + pluginID)) {
-            options = Client4.getOptions(options);
-        }
-
-        if (credentials) {
-            options.credentials = credentials;
-        }
+    doGet = (url, headers = {}, credentials) => {
+        return async (dispatch, getState) => {
```

Comment:
Don't we need to `dispatch` these functions since they are returning [thunks](https://github.com/reduxjs/redux-thunk#motivation) now?

As in, anywhere we are calling `doGet(url)`, we need to change this to be `dispatch(doGet(url))`. 1/5 Maybe we should go the route of "setting" the site URL on the client as we have done in similar PRs for other plugins.

Reply:
@mickmister I agree with what you mean regards to the setup of the site url but the implementation is different on many of the plugins, can you tell me which plugin I should follow

Reply:
Let's go with the implementation from https://github.com/mattermost/mattermost-plugin-github/pull/459/files